zulipterminal/ui_tools/boxes.py

```diff
@@ -954,39 +954,29 @@ def soup2markup(cls, soup: Any, metadata: Dict[str, Any],
                     metadata['bq_len'] -= 1
                     continue
                 markup.append(element)
-            elif tag == 'div' and (
-                 set(tag_classes) & set(unrendered_div_classes)):
-                # UNRENDERED DIV CLASSES
-                # NOTE: Though `matches` is generalized for multiple
-                # matches it is very unlikely that there would be any.
-                matches = set(unrendered_div_classes) & set(tag_classes)
-                text = unrendered_div_classes[matches.pop()]
-                if text:
-                    markup.append(unrendered_template.format(text))
-            elif tag == 'img' and tag_classes == ['emoji']:
-                # CUSTOM EMOJIS AND ZULIP_EXTRA_EMOJI
-                emoji_name = tag_attrs.get('title', [])
-                markup.append(('msg_emoji', f":{emoji_name}:"))
-            elif tag in unrendered_tags:
-                # UNRENDERED SIMPLE TAGS
-                text = unrendered_tags[tag]
-                if text:
-                    markup.append(unrendered_template.format(text))
             elif tag in ('p', 'del'):
                 # PARAGRAPH, STRIKE-THROUGH
                 markup.extend(cls.soup2markup(element, metadata)[0])
-            elif tag == 'span' and 'emoji' in tag_classes:
-                # EMOJI
-                markup.append(('msg_emoji', tag_text))
-            elif tag == 'span' and (
-                 set(('katex-display', 'katex')) & set(tag_classes)):
-                # MATH TEXT
-                markup.append(tag_text)
+            elif tag in ('strong', 'em'):
```

Comment:
Could you explain your prioritization of each tag?

Reply:
I think the ordering should be done according to most used tags/classes so that the loop ends faster, which it is not doing right now. Many of the message tags are going to be `p` tags as it is text. Since bold and italics are also text related, I have grouped them together. Next most used are `mentions` and `blockquotes` I  feel after which come emoji's. I have also grouped both the emojis together. I wanted to put Links and Images(unrendered tags) before emoji but since custom-emoji's have an `img` tag it had to be processed before. Towards the end I have put least used (time, math, twitter preview) before which I have put code, lists and tables which are moderately used. Position of `unrendered_div` is still debatable  

| Currently | Proposed |
| :----------:  | :-----------: |
| NavString | NavString |
| Unrendered Div classes | Paragraph |
| Custom emoji | Bold-Italics |
| Unrendered Tags | mentions |
| Paragraph | block quotes |
| emoji | emoji |
| math | custom emoji |
| mentions | Links |
| links | unrendered-tags(has images) |
| blockquote | code |
| code | codeblock|
| block-code | lists list-items |
| bold-italics | table |
| lists and list-items | unrendered div(inline img, inline ref, twitter) |
| table | math |
| time | time |